sources/api/certdog/src/main.rs

```diff
@@ -0,0 +1,401 @@
+#[derive(Debug)]
+struct Args {
+    log_level: LevelFilter,
+    socket_path: String,
+    trusted_store: String,
+    source_bundle: String,
+    comment: bool,
+}
+
+impl Default for Args {
+    fn default() -> Self {
+        Self {
+            log_level: LevelFilter::Info,
+            socket_path: DEFAULT_API_SOCKET.to_string(),
+            trusted_store: DEFAULT_TRUSTED_STORE.to_string(),
+            source_bundle: DEFAULT_SOURCE_BUNDLE.to_string(),
+            comment: false,
+        }
+    }
+}
+
+/// Print a usage message in the event a bad arg is passed
+fn usage() {
+    let program_name = env::args().next().unwrap_or_else(|| "program".to_string());
+    eprintln!(
+        r"Usage: {} [ ARGUMENTS... ]
+
+    Global arguments:
+        [ --socket-path PATH ]
+        [ --source-bundle PATH]
+        [ --trusted-store PATH]
+        [ --comment]
+        [ --log-level trace|debug|info|warn|error ]
+
+    Socket path defaults to {}
+    Source bundle defaults to {}
+    Trusted store defaults to {}
+    ",
+        program_name, DEFAULT_API_SOCKET, DEFAULT_SOURCE_BUNDLE, DEFAULT_TRUSTED_STORE
+    );
+}
+
+/// Parses user arguments into an Args struct
+fn parse_args(args: env::Args) -> Result<Args> {
+    let mut final_args = Args::default();
+
+    let mut iter = args.skip(1);
+    while let Some(arg) = iter.next() {
+        match arg.as_ref() {
+            "--log-level" => {
+                let log_level = iter.next().context(error::Usage {
+                    message: "Did not give argument to --log-level",
+                })?;
+                final_args.log_level =
+                    LevelFilter::from_str(&log_level).context(error::LogLevel { log_level })?;
+            }
+
+            "-s" | "--socket-path" => {
+                final_args.socket_path = iter.next().context(error::Usage {
+                    message: "Did not give argument to --socket-path",
+                })?
+            }
+
+            "-t" | "--trusted-store" => {
+                final_args.trusted_store = iter.next().context(error::Usage {
+                    message: "Did not give argument to --trusted-store",
+                })?
+            }
+
+            "-b" | "--source-bundle" => {
+                final_args.source_bundle = iter.next().context(error::Usage {
+                    message: "Did not give argument to --source-bundle",
+                })?
+            }
+
+            "-c" | "--comment" => final_args.comment = true,
+
+            x => {
+                return error::Usage {
+                    message: format!("Unknown option '{}'", x),
+                }
+                .fail()
+            }
+        }
+    }
+
+    Ok(final_args)
+}
+
+/// Query the API for the certificate bundles, returns a tuple with trusted
+/// and distrusted PEM certificates
+async fn get_certificate_bundles<P>(
+    socket_path: P,
+) -> Result<(Vec<x509_parser::pem::Pem>, Vec<x509_parser::pem::Pem>)>
```

Comment:
Maybe something like this instead, with a structured type?

```suggestion
) -> Result<CertBundle>

...

struct CertBundle {
    trusted_certs: Vec<x509_parser::pem::Pem>,
    distrusted_certs: Vec<x509_parser::pem::Pem>,
}

impl TryFrom<HashMap<Identifier, model::PEMCertificate>> for CertBundle {};
```